legit.py

```diff
@@ -0,0 +1,54 @@
+from git import Repo
+from git.exc import GitCommandError
+from typing import Iterable
+
+
+class Repository:
+
+    def __init__(self, path: str) -> None:
+        self.repo = Repo.init(path)
+
+    @classmethod
+    def clone(cls, url: str, path: str) -> str:
+        try:
+            Repo.clone_from(url, path)
+        except GitCommandError:
+            return None
+        return cls(path)
+
+    @classmethod
+    def init(cls, path: str) -> str:
```

Comment:
init is a very poor choice of a method name in Python because of the special role of `__init__`.

Reply:
Thanks, I know, I've done it to stick with original naming from lib and git as well
I could change, pick name for it, please

Reply:
> bare_repo = Repo.init(os.path.join(rw_dir, 'bare-repo'), bare=True)
https://gitpython.readthedocs.io/en/stable/tutorial.html#advanced-repo-usage

Reply:
init_repo? initialize? 